### PR TITLE
twisted modules

### PR DESCRIPTION
**Pull request by wdecker**

### CODE REVIEW

File: src/Modules/ModulesGraded.jl [hunk 1]
```diff
@@ -2567,3 +2567,89 @@ function ideal_as_module(I::MPolyIdeal)
...
+    twist(M::ModuleFP{T}, d::IntegerUnion) where {T<:MPolyDecRingElem}
+
+Given a module `M` over a $\mathbb Z$-graded polynomial ring and an integer `d`, 
+convert `d` into an element `g` of the grading group of the ring and proceed as above.
+
+# Examples
+```jldoctest
+julia> R, (x, y) = graded_polynomial_ring(QQ, ["x", "y"]);
+
+julia> I =  ideal(R, [zero(R)])
```
Comment (lgoettgens) [inline src/Modules/ModulesGraded.jl]:
```suggestion
julia> I = ideal(R, [zero(R)])
```

File: src/Modules/ModulesGraded.jl [hunk 2]
```diff
@@ -2567,3 +2567,89 @@ function ideal_as_module(I::MPolyIdeal)
...
+
+```
+"""
+function twist(M::ModuleFP{T}, g::GrpAbFinGenElem) where {T<:MPolyDecRingElem}
+ error("Not implemented for the given type")
+end
+
+function twist(M::SubquoModule{T}, g::GrpAbFinGenElem) where {T<:MPolyDecRingElem}
+ R = base_ring(M)
+ @req parent(g) == grading_group(R) "Group element not contained in grading group of base ring"
```
Comment (lgoettgens) [inline src/Modules/ModulesGraded.jl:2623]: Is `==` here enough or do we need `===`? (same at other places)

File: test/Modules/ModulesGraded.jl [hunk 3]
```diff
@@ -1166,3 +1166,13 @@ end
   b = transpose(a) * a
   @test is_symmetric(b)
 end
+
+@testset "twist" begin
+  R, (x, y) = graded_polynomial_ring(QQ, ["x", "y"], [1 0; 0 1]);
+  I = ideal(R, [x, y])
+  M = quotient_ring_as_module(I)
+  N = twist(M, [1; 2])
```
Comment (lgoettgens) [inline test/Modules/ModulesGraded.jl]:
```suggestion
  N = twist(M, [1, 2])
```
I think you want a vector here instead

File: src/Modules/ModulesGraded.jl (hunk 2 above)
Comment (wdecker) [reply src/Modules/ModulesGraded.jl:2623]: @fieker Please clearify!

Comment (afkafkafk13) [reply src/Modules/ModulesGraded.jl:2623]: Sideremark: 
If `===` is needed, this problem is not restricted to this PR. The problem then also occurs in `Modules/ModulesGraded.jl` (and possibly other files) as the same kind of comparison also occurs there with `==`.

Comment (fingolfin) [reply src/Modules/ModulesGraded.jl:2623]: I think `==` and `===` are the same for rings, so either is fine.

**Comment by jankoboehm** [comment]
Thanks for spotting, it should be ```===```, and it should also be changed in ```set_grading!```, ```graded_free_module```, and ```grade```, every time the same copied line.

File: src/Modules/ModulesGraded.jl [hunk 4]
```diff
@@ -2567,3 +2567,89 @@ function ideal_as_module(I::MPolyIdeal)
...
+1 -> e[1]
+represented as subquotient with no relations
+
+julia> degree(gen(N, 1))
+[-2]
+
+```
+"""
+function twist(M::ModuleFP{T}, g::GrpAbFinGenElem) where {T<:MPolyDecRingElem}
+ error("Not implemented for the given type")
```
Comment (fingolfin) [inline src/Modules/ModulesGraded.jl:2618]: I am not going to block this PR over this, but, normally we strive to indent code with (at least) two spaces.

(But of course we are already widely inconsistent about this, which is why I am not asking for it to be changed here and now, just wanted to point it out)

File: src/Modules/ModulesGraded.jl (hunk 2 above)
Comment (wdecker) [reply src/Modules/ModulesGraded.jl:2623]: As confirmed by @fieker, it is as I thought: There is no difference between `==` and `===` here. That is, the PR can be merged and no other action is needed.

**Review by afkafkafk13 — APPROVED** [review]

**Pull request merged**